File: api-gateway/src/main/java/md/vnastasi/cloud/ApplicationProperties.java
package md.vnastasi.cloud;

import org.springframework.boot.context.properties.ConfigurationProperties;
import org.springframework.boot.context.properties.ConstructorBinding;

import java.util.List;

@ConfigurationProperties(prefix = "application")
@ConstructorBinding
public record ApplicationProperties(
        Security security
) {

    public static record Security(
            String user,
            String password,
            List<String> roles
    ) {}
}

File: timetable/src/main/kotlin/md/vnastasi/cloud/ApplicationProperties.kt
package md.vnastasi.cloud

import org.springframework.boot.context.properties.ConfigurationProperties
import org.springframework.stereotype.Component

@Component
@ConfigurationProperties(prefix = "application")
class ApplicationProperties {

    var nsApi: NsApiProperties = NsApiProperties()
}

class NsApiProperties {

    var publicTravelInfo: PublicTravelInfoProperties = PublicTravelInfoProperties()
    lateinit var baseUrl: String
}

class PublicTravelInfoProperties {

    var apiKey: ApiKeyProperties = ApiKeyProperties()
    lateinit var basePath: String
    lateinit var arrivalsPath: String
    lateinit var departuresPath: String
}

class ApiKeyProperties {

    lateinit var header: String
    lateinit var value: String
}
